graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DataTypeMapping.java

```diff
@@ -70,47 +135,62 @@ static Object toDbValue(Column.ColumnType type, Object value) {
     Map<String, Object> map = new HashMap<>(columns.size());
     for (Column column : columns) {
       if (row.has(column)) {
-        map.put(nameMapping.getColumnName(table).get(column), toGraphQLValue(column, row));
+        map.put(
+            nameMapping.getColumnNames(table).get(column),
+            toGraphQLValue(nameMapping, column, row));
       }
     }
     return map;
   }
 
-  private static Object toGraphQLValue(Column column, Row row) {
+  private static Object toGraphQLValue(NameMapping nameMapping, Column column, Row row) {
     Object dbValue = row.getValue(column.name());
-    return toGraphQLValue(column.type(), dbValue);
+    return toGraphQLValue(nameMapping, column.type(), dbValue);
   }
 
-  private static Object toGraphQLValue(Column.ColumnType type, Object dbValue) {
-    if (!type.isCollection() || dbValue == null) {
+  private static Object toGraphQLValue(
+      NameMapping nameMapping, Column.ColumnType type, Object dbValue) {
+    if (dbValue == null) {
+      return null;
+    } else if (type.isCollection()) {
+      if (type.rawType() == Column.Type.List || type.rawType() == Column.Type.Set) {
+        Collection<?> dbCollection = (Collection<?>) dbValue;
+        return dbCollection.stream()
+            .map(item -> toGraphQLValue(nameMapping, type.parameters().get(0), item))
+            .collect(Collectors.toList());
```

Comment:
The shortcuts for non-nested collections don't work anymore, we need to always recurse into elements to properly handle UDTs.